include/avif/avif.h

```diff
@@ -701,6 +705,11 @@ typedef struct avifDecoder
     avifBool ignoreExif;
     avifBool ignoreXMP;
 
+    // This represents the maximum size of a image (in pixel count) that the underlying AV1 decoder
+    // should attempt to decode. It defaults to AVIF_MAX_IMAGE_SIZE, and can be set to 0 to disable
+    // the limit. Currently supported codecs: dav1d.
+    uint32_t imageSizeLimit;
```

Comment:
`imageSizeLimit` is not a member of `avifImage`, so bullets 1 and 2 don't apply here, right?

Reply:
Sorry about my confusion. I checked the wrong function. Bullet 1 should read:

> 1. In src/read.c, avifDecoderCreate() needs to initialize imageSizeLimit to the default (AVIF_MAX_IMAGE_SIZE).

Bullet 2 is correct. I was referring to the following code in avifParseImageGridBox():

```
    if ((grid->outputWidth == 0) || (grid->outputHeight == 0) || (grid->outputWidth > (AVIF_MAX_IMAGE_SIZE / grid->outputHeight))) {
        return AVIF_FALSE;
    }
```

Reply:
Ah, I see. That is a good question.

Do we want `imageSizeLimit` to be something solely enforced by the underlying AV1 decoders, or do we want to explicitly check these limits even at the container level? For example, if the `ispe` box returned something that exceeds `imageSizeLimit`, do we bail out immediately? Do we want a new error of `AVIF_RESULT_IMAGE_TOO_LARGE`?

Reply:
imageSizeLimit serves two purposes:

1. To help prevent integer arithmetic overflow in libavif client's code. For this purpose, we want to explicitly check the image size limits even at the container level. (I believe we are not doing that now.) I think we should add a new error of `AVIF_RESULT_IMAGE_TOO_LARGE`.
2. oss-fuzz tests fail if we allocate too much memory from the heap. (oss-fuzz doesn't care whether we handle out-of-memory errors properly.) For this purpose, we don't need to check the image size limits at the container level, but we do need to check the grid image sizes (because we will call `avifImageAllocatePlanes()` in `avifDecoderDataFillImageGrid()`), and we also need the underlying AV1 decoders to enforce the same size limits.